src/shared/components/mutationTable/column/ascnCopyNumber/ASCNCopyNumberColumnFormatter.tsx

```diff
@@ -0,0 +1,195 @@
+        );
+        if (
+            ascnCopyNumberValue !== ASCNCopyNumberValue.NA &&
+            hasASCNProperty(mutation, 'totalCopyNumber') &&
+            getWGD(sampleIdToClinicalDataMap, mutation.sampleId) !==
+                ASCNCopyNumberValue.NA &&
+            getASCNCopyNumberColor(ascnCopyNumberValue.toString()) !==
+                ASCN_BLACK
+        ) {
+            sampleToCNA[
+                mutation.sampleId
+            ] = mutation.alleleSpecificCopyNumber.totalCopyNumber.toString();
+        } else {
+            sampleToCNA[mutation.sampleId] = ASCNCopyNumberValue.NA;
+        }
+    }
+    return sampleToCNA;
+}
+
+function getSortValue(
+    data: Mutation[],
+    sampleIdToClinicalDataMap: { [key: string]: ClinicalData[] } | undefined,
+    sampleIds: string[]
+) {
+    const displayValuesBySample: {
+        [key: string]: string;
+    } = getAllTotalCopyNumberForMutation(
+        data,
+        sampleIdToClinicalDataMap,
+        sampleIds
+    );
+    const sampleIdsWithValues = sampleIds.filter(
+        sampleId => displayValuesBySample[sampleId]
+    );
+    const displayValuesAsString = sampleIdsWithValues.map(
+        (sampleId: string) => {
+            return displayValuesBySample[sampleId];
+        }
+    );
+    return displayValuesAsString.join(';');
+}
+
+export function getWGD(
+    sampleIdToClinicalDataMap:
+        | { [sampleId: string]: ClinicalData[] }
+        | undefined,
+    sampleId: string
+) {
+    let wgdData = sampleIdToClinicalDataMap
+        ? sampleIdToClinicalDataMap[sampleId].filter(
+              (cd: ClinicalData) => cd.clinicalAttributeId === 'FACETS_WGD'
+          )
+        : undefined;
+    return wgdData !== undefined && wgdData.length > 0
+        ? wgdData[0].value
+        : ASCNCopyNumberValue.NA;
+}
+
+export const getDefaultASCNCopyNumberColumnDefinition = (
+    sampleIds?: string[],
+    sampleIdToClinicalDataMap?: { [sampleId: string]: ClinicalData[] },
+    sampleManager?: SampleManager
+) => {
+    return {
+        name: 'Integer Copy #',
+        render: (d: Mutation[]) =>
+            ASCNCopyNumberColumnFormatter.renderFunction(
+                d,
+                sampleIds ? sampleIds : d.length > 0 ? [d[0].sampleId] : [],
+                sampleIdToClinicalDataMap,
+                sampleManager
+            ),
+        sortBy: (d: Mutation[]) =>
+            getSortValue(
+                d,
+                sampleIdToClinicalDataMap,
+                sampleIds ? sampleIds : d.length > 0 ? [d[0].sampleId] : []
+            ),
+    };
+};
+
+export default class ASCNCopyNumberColumnFormatter {
+    /* Determines the display value by using the impact field.
+     *
+     * @param data  column formatter data
+     * @returns {string}"Clonal" text value
+     */
+    public static renderFunction(
+        data: Mutation[],
+        sampleIds: string[],
+        sampleIdToClinicalDataMap?: { [sampleId: string]: ClinicalData[] },
```

Comment:
how come this is optional?

Reply:
@sheridancbio probably fine

Reply:
We think it is possible for a study to render with ASCN data, but without Whole Genome Duplication data. In this case sampleIdToClinicalDataMap might be undefined and that would cause the page to crash.